framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceImplementation.java

```diff
@@ -1001,6 +1001,10 @@ protected void updateAnnotationSet(
       return;
     }
 
+    // Update Initialization Annotations in WPI. It is only applied to
+    // Nullness-related qualifiers' inference to prevent possible loops.
```

Comment:
This comment is inappropriate, because it may not remain true. In particular, it is mixing implementation details of a specific checker (nullness) with the generic WPI algorithm (which is what this code implements). The comment should explain the general purpose of this callback hook, not the specific use in one checker.